eip/EIPS/eip-aa-rules.md

```diff
@@ -0,0 +1,351 @@
+---
+eip: 0
+title: Account Abstraction Validation Scope Rules
+description: A set of limitations on validation EVM code that protects Account Abstraction nodes from exponentially complex computations without compensation
+author: Yoav Weiss (@yoavw), Dror Tirosh (@drortirosh), Alex Forshtat (@forshtat), Shahaf Nacson (@shahafn)
+discussions-to: https://www.google.com/
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-01
+requires: 6780
+---
+
+## Abstract
+
+This document describes the validation rules we impose on the validation context of Account Abstraction transactions,
+such as [ERC-4337](./eip-4337) `UserOperation` or [EIP-9999](./eip-9999), which are enforced off-chain by a
+block builder or a standalone bundler, and the rationale behind each one of them.
+
+## Motivation
+
+All transactions initiated by EOAs initiated have an implicit validation phase where balance, nonce, and signature are
+checked to be valid for the current state of the Ethereum blockchain.
+Once the transaction is checked to be valid by a node, only another transaction by the same EOA can modify the Ethereum
+state in a way that makes the first transaction invalid.
+
+With Account Abstraction, however, the validation can also include an arbitrary EVM code and rely on storage as well,
+which means that unrelated `UserOperations` or transactions may invalidate each other.
+
+If not addressed, this would make the job of maintaining a mempool of valid `UserOperations` and producing valid
+bundles computationally infeasible and susceptible to DoS attacks.
+
+This document describes set of validation rules that, if applied by a block builder prior including a `UserOperation`,
+can prevent such attacks.
```

Comment:
> if applied by a block builder prior including

Should this be: "if applied by a block builder prior to including a `UserOperation` in the mempool"

Reply:
reverted to bundler...